src/__tests__/ApolloClient.ts

```diff
@@ -2174,6 +2174,177 @@ describe("ApolloClient", () => {
+      const ItemFragment = gql`
+        fragment ItemFragment on Item {
+          id
+          text
+        }
+      `;
+
+      cache.writeFragment({
+        fragment: ItemFragment,
+        data: {
+          __typename: "Item",
+          id: 5,
+          text: "Item #5",
+        },
+      });
+
+      const observable = client.watchFragment({
+        fragment: ItemFragment,
+        from: { __typename: "Item", id: 5 },
+      });
+
+      const stream = new ObservableStream(observable);
+
+      {
+        const result = await stream.takeNext();
+
+        expect(result).toEqual({
+          data: {
+            __typename: "Item",
+            id: 5,
+            text: "Item #5",
+          },
+          complete: true,
+        });
+      }
+    });
+    it("if only partial data is available, `complete` is `false`", async () => {
+      const cache = new InMemoryCache();
+      const client = new ApolloClient({
+        cache,
+        link: ApolloLink.empty(),
+      });
+      const ItemFragment = gql`
+        fragment ItemFragment on Item {
+          id
+          text
+        }
+      `;
+
+      {
+        // we expect a "Missing field 'text' while writing result..." error
+        // when writing item to the cache, so we'll silence the console.error
+        using _consoleSpy = spyOnConsole("error");
+        cache.writeFragment({
+          fragment: ItemFragment,
+          data: {
+            __typename: "Item",
+            id: 5,
+          },
+        });
+      }
+
+      const observable = client.watchFragment({
+        fragment: ItemFragment,
+        from: { __typename: "Item", id: 5 },
+      });
+
+      const stream = new ObservableStream(observable);
+
+      {
+        const result = await stream.takeNext();
+
+        expect(result).toEqual({
+          data: {
+            __typename: "Item",
+            id: 5,
+          },
+          complete: false,
+          missing: {
+            text: "Can't find field 'text' on Item:5 object",
+          },
+        });
+      }
+    });
+    it("if no data is written after observable is subscribed to, next is never called", async () => {
+      const cache = new InMemoryCache();
+      const client = new ApolloClient({
+        cache,
+        link: ApolloLink.empty(),
+      });
+      const ItemFragment = gql`
+        fragment ItemFragment on Item {
+          id
+          text
+        }
+      `;
+
+      const observable = client.watchFragment({
+        fragment: ItemFragment,
+        from: { __typename: "Item", id: 5 },
+      });
+
+      const stream = new ObservableStream(observable);
+
+      {
+        const result = await stream.takeNext();
+
+        expect(result).toEqual({
+          complete: false,
+          data: {},
+          missing: "Dangling reference to missing Item:5 object",
+        });
+      }
+
+      const handleNext = jest.fn();
+
+      observable.subscribe(handleNext);
+
+      expect(handleNext).toHaveBeenCalledTimes(0);
```

Comment:
I'd try not to create a new subscription here, since new subscriptions can sometimes "kick off" new behaviour and we don't want to test that.

How about something like this?
```suggestion
  await expect(stream.takeNext({timeout: 1000})).rejects.toEqual(expect.any(Error));
```

Reply:
Resolved in [4a8dac0](https://github.com/apollographql/apollo-client/pull/11465/commits/4a8dac064f2254cfe94cf74633d1130cfb52f9a4)